swap/src/asb/event_loop.rs

```diff
@@ -322,10 +322,30 @@ where
         let rate = self
             .latest_rate
             .latest_rate()
-            .context("Failed to get latest rate")?;
+            .context("Failed to get latest rate")?
+            .ask()
+            .context("Failed to compute asking price")?;
+
+        let max_bitcoin_for_monero = self
+            .monero_wallet
+            .get_balance()
+            .await?
+            .max_bitcoin_from_balance(rate.as_sat())?;
```

Comment:
Why not pass the `bitcoin::Amount` directly? Gives you more type-safety! :)

Reply:
Done!